Homework 7

mrnchernova opened the pull request.

Commit:
homework 7: Math, Arrays, Date

Commit:
homework 7: Performance appraisal  String, StringBuilder and StringBuffer

Commit:
homework 7: Search for hexadecimal numbers

Commit:
homework 7: Search for tags (not done)

Commit:
homework 7: Search for tags

RomanKashenok commented on part_1/src/com/chernova/homework_7/NotebookSortByDate.java:
```diff
@@ -0,0 +1,72 @@
+        System.out.println("Запишите задачу или нажмите \".\", если хотите перейти к списку");
+
+        s = sc.nextLine();
+        while (!s.equals(".")) {
```
если ты хочешь работать с вводом пользователя, то уже давай полностью в дальнейшем строки обрабатывать (обрезать пробелы в начале\конце. Проверять что ввод валиден )) 
Но в целом все норм

mrnchernova replied:
Добавила проверку на спецсимволы. 
Теперь пробелы в начале и в конце строки будут обрезаны

RomanKashenok commented on part_1/src/com/chernova/homework_7/NotebookSortByDate.java:
```diff
@@ -0,0 +1,72 @@
+
+public class NotebookSortByDate {
+    public static void main(String[] args) {
+        ArrayList<Notebook> notes = new ArrayList<Notebook>();
```
вот это не совсем ок. Точнее ок, но класс назови Note, а Notebook - это пусть уже сама записная книжка будет

RomanKashenok commented on part_1/src/com/chernova/homework_7/NotebookSortByDate.java:
```diff
@@ -0,0 +1,72 @@
+            System.out.println(i);
+        }
+
+        Collections.sort(notes);
```
все хорошо и правильно, но цель задачи все же - научиться работать и писать алгоритмы )) 
Пользоваться готовым - это классно и хорошо, что знаешь как. Но на собеседованиях, как правило, просят написать алгоритм той же сортировки

mrnchernova replied:
Переделано. Теперь записи сортируются алгоритмом сортировки вставками

RomanKashenok commented on part_1/src/com/chernova/homework_7/Task21StringPerformanceAppraisal.java:
```diff
@@ -0,0 +1,39 @@
+        Long ld1,ld2;
+
+        ld1 = System.currentTimeMillis();
+        for (int i = 0; i < 100000 ; i++) s = s + i;
```
давай в фигурные скобки область видимости засовывать

RomanKashenok reviewed (approved):
Есть вопросики, но все норм в целом

Commit:
homework 7: Task21StringPerformanceAppraisal.java
Added braces for loop

Commit:
homework 7: TaskNotebookSortByDate.java
- added sorting
- added string input check
- changed class name from Notebook to Note

mrnchernova reviewed (commented):
fixed remarks
